client/lib/cart/store/cart-synchronizer.js

```diff
@@ -106,17 +107,27 @@ CartSynchronizer.prototype.update = function( changeFunction ) {
 		// with a consistent state when we apply our local changes. The strategy
 		// here is to queue the changes until the first request has completed, then
 		// proceed as normal.
-		this._enqueueChange( changeFunction );
+		this._enqueueChange( changeFunction, serverFlushCallback );
+		if ( ! this._activeRequest ) {
+			this.fetch();
+		}
+		debug( 'update: change enqueued' );
 		return;
 	}
 
+	debug( 'update: processing change' );
+	this._addFlushCallback( serverFlushCallback );
+	this._processChangeFunction( changeFunction );
+	this.emit( 'change' );
+};
+
+CartSynchronizer.prototype._processChangeFunction = function ( changeFunction ) {
 	if ( this._activeRequest && this._activeRequest.state !== 'canceled' ) {
 		this._activeRequest.state = 'canceled';
 	}
 
 	this._latestValue = changeFunction( this._latestValue );
 	this._performRequest( 'update', this._postToServer.bind( this ) );
-	this.emit( 'change' );
```

Comment:
This is git playing games, `this.emit( 'change' )` is at the same place it was before, at the end of `update()` function.

> Why not emit events instead of queuing callbacks?

currently we call it threw an actionCreator, like so:
```js
addItems( [ domainItem ], 'signup', error => callback( error, providedDependencies ) );
```

So i'm not sure how emitting events will work, could you maybe show an example?